ingeniamotion/comkit.py

```diff
@@ -0,0 +1,189 @@
+import io
+import xml.etree.ElementTree as ET
+from xml.etree.ElementTree import ElementTree, Element
+from enum import Enum
+from typing import Optional
+from xml.dom import minidom
+import tempfile
+from pathlib import Path
+
+FILE_EXT_DICTIONARY = ".xdf"
+DICT_DEVICE_PRODUCT_CODE = "ProductCode"
+DICT_DEVICE_PART_NUMBER = "PartNumber"
+DICT_DEVICE_PART_NUMBER_COCO = "PartNumberCoco"
+DICT_DEVICE_PART_NUMBER_MOCO = "PartNumberMoco"
+
+MOCO_SUBNODE = 1
+SUBNODE_ATTRIBUTE = "subnode"
+REGISTER_SECTION = "Body/Device/Registers"
+DEVICE_SECTION = "Body/Device"
+ERRORS_SECTION = "Body/Errors"
+IMAGE_SECTION = "DriveImage"
+
+
+class CORE(Enum):
+    """Core type enum"""
+
+    COMMUNICATION_CORE = "Coco"
+    MOTION_CORE = "Moco"
+
+
+def create_comkit_dictionary(
+    coco_dict_path: str, moco_dict_path: str, dest_folder: Optional[str] = None
+) -> str:
+    """Create a dictionary for COMKIT by merging a COCO dictionary and a MOCO dictionary.
+
+    Args:
+        coco_dict_path : COCO dictionary path.
+        moco_dict_path : MOCO dictionary path.
+        dest_folder: Path to a folder to store the COMKIT dictionary. If it's not provided the
+            merged dictionary is stored in the temporary system's folder.
+
+    Returns:
+        Path to the COMKIT dictionary.
+
+    """
+    coco_dict_name = Path(coco_dict_path).stem
+    moco_dict_name = Path(moco_dict_path).stem
+    if dest_folder is None:
+        dest_folder = tempfile.gettempdir()
+    dest_path = f"{dest_folder}/{coco_dict_name}-{moco_dict_name}{FILE_EXT_DICTIONARY}"
```

Comment:
I think generate this filename is not necessary. You can add the filename as an optional parameter and use a temporary file if filename is None, with a random name.

Reply:
Done!